### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by VikingScientist**

### CODE REVIEW

File: splipy/BSplineBasis.py [hunk 1]
```diff
@@ -446,6 +447,23 @@ def matches(self, bspline, reverse=False):
...
+    def snap(self, t):
+        """  Snap evaluation points to knots if they are sufficiently close
+        as given in by state.state.knot_tolerance. This will modify the input vector t
+
+        :param t: evaluation points
+        :type  t: [float]
+        :return: none
+        """
+        n = len(self.knots)
+        for j in range(len(t)):
```
Comment (TheBB) [inline splipy/BSplineBasis.py:459]: If the only purpose of `j` is to index into `t` then just `for tt in t` or similar.

File: splipy/BSplineBasis.py [hunk 2]
```diff
@@ -446,6 +447,23 @@ def matches(self, bspline, reverse=False):
...
+            i = bisect_left(self.knots, t[j])
+            if( i < n and abs(self.knots[i]-t[j]) < state.knot_tolerance):
```
Comment (TheBB) [inline splipy/BSplineBasis.py]: Don't need to parenthesize conditions.

File: splipy/basis_test.py [hunk 3]
```diff
@@ -228,5 +228,33 @@ def test_make_periodic(self):
...
+        self.assertNotEqual(t[4], b[4]) # 3.14
+        self.assertNotEqual(t[5], b[4]) # 3.1415
+        self.assertNotEqual(t[6], b[4]) # 3.1415926
+        self.assertEqual(   t[7], b[5]) # 3.5
+        self.assertEqual(   t[8], b[6]) # 4.0
+        self.assertNotEqual(t[9], b[6]) # 4.01
+
+    def test_bug44(self):
+        # https://github.com/sintefmath/Splipy/issues/44
+        b = BSplineBasis(4, [ 0.        , 0.        , 0.        , 0.        , 0.01092385, 0.02200375,\
```
Comment (TheBB) [inline splipy/basis_test.py]: Python has implicit line-continuation inside bracket pairs, `()`, `[]` and `{}`. So no need for a backslash here.

File: splipy/basis_test.py [hunk 4]
```diff
@@ -228,5 +228,33 @@ def test_make_periodic(self):
...
+                              0.03316167, 0.04435861, 0.05526295, 0.0663331 , 0.07748615, 0.08868065,\
+                              0.09989588, 0.11080936, 0.12188408, 0.13303942, 0.14423507, 0.15545087,\
+                              0.16636463, 0.1774395 , 0.1885949 , 0.19979059, 0.2       , 0.2       ,\
+                              0.2       , 0.2     ]);
```
Comment (TheBB) [inline splipy/basis_test.py]: No need for a semicolon either for that matter. :-P

**Review by TheBB — CHANGES_REQUESTED** [review]
Looks fine, just some anti-Pythonisms.

File: splipy/BSplineBasis.py (hunk 1 above)
Comment (VikingScientist) [reply splipy/BSplineBasis.py:459]: I also need to change the content of `t` (see line 462) and then your proposed syntax won't work.

**Comment by VikingScientist** [comment]
Updated PR with all requested changes. Let me know if it's OK so I can run `git rebase --autosquash` before merging.

**Comment by TheBB** [comment]
Thanks, looks fine to me. Feel free to merge yourself, I'm on the phone.

**Pull request merged**